app/services/katello/pulp/repository.rb

```diff
@@ -120,6 +121,55 @@ def mirror_importer_connection_options
       def proxy_host_importer_value
         root.ignore_global_proxy ? "" : nil
       end
+
+      def refresh
+        update_or_associate_importer
+        update_or_associate_distributors
+        remove_unnecessary_distributors
+      end
+
+      def update_or_associate_importer
+        existing_importers = backend_data["importers"]
+        importer = generate_importer
+        found = existing_importers.find { |i| i['importer_type_id'] == importer.id }
+
+        if found
+          ssl_ca_cert = importer.config.delete('ssl_ca_cert')
+          ssl_client_cert = importer.config.delete('ssl_client_cert')
+          ssl_client_key = importer.config.delete('ssl_client_key')
+          importer.config['basic_auth_username'] = nil if importer.config['basic_auth_username'].blank?
+          importer.config['basic_auth_password'] = nil if importer.config['basic_auth_password'].blank?
+          # Update ssl options by themselves workaround for https://pulp.plan.io/issues/2727
+          smart_proxy.pulp_api.resources.repository.update_importer(repo.pulp_id, found['id'], :ssl_client_cert => ssl_client_cert,
```

Comment:
@sjha4  @jlsherrill its seems that many of this calls are asynchronous: was it intentional that we're not waiting for them to finish anymore?

Reply:
Thanks @iNecas i opened a pr here: https://github.com/Katello/katello/pull/7968

Reply:
:+1: